dm/master/shardddl/pessimist.go

```diff
@@ -67,14 +68,65 @@ func (p *Pessimist) Start(pCtx context.Context, etcdCli *clientv3.Client) error
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
+	rev1, rev2, err := p.buildLocks(etcdCli)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithCancel(pCtx)
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+		// TODO: handle fatal error from run
+		p.run(ctx, etcdCli, rev1, rev2)
+	}()
+
+	p.closed = false // started now.
+	p.cancel = cancel
+	p.cli = etcdCli
```

Comment:
I think you should update `p.cli = etcdCli` both for L85 and L122.

Reply:
updated in https://github.com/pingcap/dm/pull/555/commits/63fa07b3117e3fe07b7344108ec55708cb168c73